src/Network.cpp

```diff
@@ -380,7 +380,17 @@ void Network::select_precision(int channels) {
 
         myprintf("Initializing OpenCL (autodetecting precision).\n");
 
-        // Start by setting up fp16.
+        // Start by setting up fp32.
+        try {
+            m_forward.reset();
+            m_forward = init_net(channels,
+                std::make_unique<OpenCLScheduler<float>>());
+            score_fp32 = benchmark_time(100);
+        } catch (...) {
+            // empty - if exception thrown just throw away fp32 net
+        }
+
+        // Now benchmark fp16.
```

Comment:
The original code simply skipped all benchmarking if we have fp16 compute.  Now we run fp32 even if we have fp16 compute so it's slightly different here.  Suggest we test fp16 --> test fp32 --> re-create fp16 if fp16 is better so that we can avoid testing fp32 at all?

Reply:
That logic seems to go against #1873, as it seems that #1873 explicitly tries to optimize for fp16 compute GPUs.  It seemed that it was intentional to test fp16 first.  @gcp  - comments?

Reply:
The whole idea of that change was to avoid doing any benchmarking if we can decide immediately fp16 is going to be faster.